adot-testbed/app/src/test/java/software/amazon/adot/testbed/LogsTests.java

```diff
@@ -65,61 +73,164 @@ private GenericContainer<?> createAndStartCollector(String configFilePath, Strin
+        collector = createAndStartCollector("/configurations/config-rfcsyslog.yaml", logStreamName);
 
-        validateLogs(logStreamName , "/logs/RFC5424.log");
+        List<String> logFilePaths = new ArrayList<>();
+        logFilePaths.add("/logs/RFC5424.log");
+        validateLogs(logStreamName , logFilePaths);
         collector.stop();
     }
 
     @Test
     void testLog4j() throws Exception {
         String logStreamName = "log4j-logstream-" + uniqueID;
-        collector = createAndStartCollector("/configurations/config-log4j.yaml", "/logs/log4j.log", logStreamName);
+        collector = createAndStartCollector("/configurations/config-log4j.yaml", logStreamName);
 
-        validateLogs(logStreamName , "/logs/log4j.log");
+        List<String> logFilePaths = new ArrayList<>();
+        logFilePaths.add("/logs/log4j.log");
+        validateLogs(logStreamName , logFilePaths);
         collector.stop();
     }
 
     @Test
     void testJson() throws Exception {
         String logStreamName = "json-logstream-" + uniqueID;
-        collector = createAndStartCollector("/configurations/config-json.yaml", "/logs/testingJSON.log", logStreamName);
+        collector = createAndStartCollector("/configurations/config-json.yaml", logStreamName);
+
+        List<String> logFilePaths = new ArrayList<>();
+        logFilePaths.add("/logs/testingJSON.log");
+        validateLogs(logStreamName , logFilePaths);
+        collector.stop();
+    }
+
+    @Test
+    void testCollectorRestartStorageExtension() throws Exception {
+        String logStreamName = "storageExtension-logstream-" + uniqueID;
+        collector = createAndStartCollector("/configurations/config-storageExtension.yaml", logStreamName);
+        File tempFile = new File(logDirectory.toString(), "storageExtension.log");
+        Thread.sleep(5000);
+
+        FileWriter fileWriter = new FileWriter(tempFile);
+        try {
+            fileWriter.write("First Message, collector is running" + "\n");
+            fileWriter.write("Second Message, collector is running" + "\n");
+            fileWriter.write("Third Message,  collector is running" + "\n");
+        } catch (IOException e) {
+            throw new RuntimeException("Error writing to File A.", e);
+        }
+        fileWriter.flush();
+
+        List<String> logFilePaths = new ArrayList<>();
+        String expectedLogPath = logDirectory.toString();
+        logFilePaths.add(expectedLogPath + "/storageExtension.log");
+
+        validateLogs(logStreamName , logFilePaths);
 
-        validateLogs(logStreamName , "/logs/testingJSON.log");
+        collector.stop();
+
+        // write to the file when collector is stopped
+        try {
+            fileWriter.write("First Message after collector is stopped" + "\n");
+            fileWriter.write("Second Message after the collector is stopped" + "\n");
+            fileWriter.write("Third Message after the collector is stopped" + "\n");
+            fileWriter.flush();
```

Comment:
Why is flush called inside the try block here but outside of it on line 144?

Reply:
Actually no specific reason, modified .